code/modules/antagonists/traitor/datum_traitor.dm

```diff
@@ -166,7 +165,7 @@
 			kill_objective.find_target()
 			add_objective(kill_objective)
 	else
-		if(prob(15) && !(locate(/datum/objective/download) in objectives) && !(owner.assigned_role in list(JOB_NAME_RESEARCHDIRECTOR, JOB_NAME_SCIENTIST, JOB_NAME_ROBOTICIST)))
+		if(prob(15) && !(locate(/datum/objective/download) in objectives) && !owner.has_job(GLOB.science_positions+JOB_KEY_GENETICIST))
```

Comment:
geneticist is no longer a science position nor has access for it, so remove it here